calour/tests/test_export_html.py

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--,  Calour development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import main
+from tempfile import mkdtemp
+from os.path import join
+import shutil
+
+from calour._testing import Tests
+import calour as ca
+
+
+class ExportHtmlTests(Tests):
+    def setUp(self):
+        super().setUp()
+        self.test1 = ca.read(self.test1_biom, self.test1_samp, normalize=None)
+
+    def test_export_html(self):
+        exp = self.test1
+        d = mkdtemp()
+        f = join(d, 'calour_interactive_heatmap.html')
+        exp.export_html(sample_field='group', output_file=f, feature_field='taxonomy', title='the heatmap')
+        # load the results
+        with open(f) as output_fl:
+            output_data = output_fl.readlines()
```

Comment:
https://docs.python.org/3.6/library/tempfile.html#tempfile.NamedTemporaryFile is better than creating temp dir and then a file inside it and deleting it at the end.

Reply:
alright. didn't notice you open it in the export_html function too.